From khali@linux-fr.org Mon Oct 17 14:16:31 2005
Date: Mon, 17 Oct 2005 23:12:36 +0200
From: Jean Delvare <khali@linux-fr.org>
To: Greg KH <greg@kroah.com>
Cc: Jiri Slaby <lnx4us@gmail.com>, Deepak Saxena <dsaxena@plexity.net>
Subject: [PATCH 12/14] i2c: kzalloc cleanups, 1 of 2
Message-Id: <20051017231236.111e196d.khali@linux-fr.org>
Content-Disposition: inline; filename=kzalloc-04-drop-useless-casts.patch

Drop useless casts on kzalloc returned values, as suggested by
Jiri Slaby.

Signed-off-by: Jean Delvare <khali@linux-fr.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/i2c/busses/i2c-keywest.c |    2 +-
 drivers/i2c/busses/i2c-nforce2.c |    3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

--- gregkh-2.6.orig/drivers/i2c/busses/i2c-keywest.c
+++ gregkh-2.6/drivers/i2c/busses/i2c-keywest.c
@@ -535,7 +535,7 @@ create_iface(struct device_node *np, str
 
 	tsize = sizeof(struct keywest_iface) +
 		(sizeof(struct keywest_chan) + 4) * nchan;
-	iface = (struct keywest_iface *) kzalloc(tsize, GFP_KERNEL);
+	iface = kzalloc(tsize, GFP_KERNEL);
 	if (iface == NULL) {
 		printk(KERN_ERR "i2c-keywest: can't allocate inteface !\n");
 		pmac_low_i2c_unlock(np);
--- gregkh-2.6.orig/drivers/i2c/busses/i2c-nforce2.c
+++ gregkh-2.6/drivers/i2c/busses/i2c-nforce2.c
@@ -313,8 +313,7 @@ static int __devinit nforce2_probe(struc
 	int res1, res2;
 
 	/* we support 2 SMBus adapters */
-	if (!(smbuses = (void *)kzalloc(2*sizeof(struct nforce2_smbus),
-				       	GFP_KERNEL)))
+	if (!(smbuses = kzalloc(2*sizeof(struct nforce2_smbus), GFP_KERNEL)))
 		return -ENOMEM;
 	pci_set_drvdata(dev, smbuses);
 
